yktrace/src/hwt/mapper.rs

```diff
@@ -108,6 +108,61 @@ impl HWTMapper {
         }
     }
 
+    /// Collapse consecutive control point blocks (including any unmappable internals) into one
+    /// control point entry. Unmappable holes exist because the generated control point calls out to
+    /// Rust code for which we have no IR.
+    ///
+    /// For example, if the trace is:
+    ///
+    /// ```ignore
+    /// [bb1, bb2, bb3, ctrlp, ctrlp, ?, ctrlp, bb4]
+    /// ```
+    ///
+    /// (where `bbN` is a mappable block, `ctrlp` is a block in the control point, and `?` is
+    /// unmappable blocks), then this function would canonicalise the trace to:
+    ///
+    /// ```ignore
+    /// [bb1, bb2, bb3, ctrlp, bb4]
+    /// ```
+    fn canonicalise_control_point(irblocks: &mut Vec<IRBlock>) {
+        let ctrlp_sym_c = CString::new(CONTROL_POINT_SYM).unwrap();
+        assert!(irblocks[0].func_name == ctrlp_sym_c);
+        let mut in_ctrlp = true;
+        let mut ctrlp_start_idx = Some(0);
```

Comment:
This doesn't need to be an `Option` AFAICS: we always assign `Some(...)` to it?

Reply:
Fixed in f9c355c.